src/ol_infrastructure/components/services/k8s.py

```diff
@@ -77,11 +81,33 @@ class OLApplicationK8sConfiguration(BaseModel):
     application_security_group_name: Output[str]
     application_service_account_name: str | Output[str] | None = None
     application_image_repository: str
-    application_image_repository_suffix: Optional[str] = None
+    application_image_repository_suffix: str | None = None
     application_docker_tag: str
-    application_cmd_array: Optional[list[str]] = None
+    application_cmd_array: list[str] | None = None
     vault_k8s_resource_auth_name: str
-    hpa_scaler_type: HPAScalerType = Field(default=HPAScalerType.CPU)
+    # You get CPU and memory autoscaling by default
+    hpa_scaling_metrics: list[kubernetes.autoscaling.v2.MetricSpecArgs] = [
+        kubernetes.autoscaling.v2.MetricSpecArgs(
+            type="Resource",
+            resource=kubernetes.autoscaling.v2.ResourceMetricSourceArgs(
+                name="CPU",
+                target=kubernetes.autoscaling.v2.MetricTargetArgs(
+                    type="Utilization",
+                    average_utilization=80,  # Target memory utilization (80%)
```

Comment:
```suggestion
                    average_utilization=80,  # Target CPU utilization (80%)
```
Assuming this is copy 🍝